middle_end/flambda/un_anf.ml

```diff
@@ -114,104 +152,98 @@ let make_var_info (clam : Clambda.ulambda) : var_info =
           ignore_int arity;
           ignore_params_with_value_kind params;
           ignore_value_kind return;
-          loop body;
+          loop ~depth:(depth + 1) body;
```

Comment:
I was wondering why function bodies where traversed in non-constant closures but not in constant ones, and I think the non-traversal of constant closures is a mistake.
In any case, since functions are not allowed to refer to variables that are not their parameters or defined inside their bodies at this point, any value of `depth` including zero should be correct.

Reply:
The comment on the constant closures case was out-of-date in that it was referencing the wrong function. I've updated it to refer to the function which actually calls un-anf on constant closures. That function does exist though, so the existing code is fine.

Any value would be correct here, but I would argue that `depth + 1` is "more correct". If functions weren't closed then the code would still be correct with `depth + 1`. No need to rely on an invariant if we don't need to.